hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/ITestReadBufferManager.java

```diff
@@ -77,14 +77,7 @@ public void testPurgeBufferManagerForParallelStreams() throws Exception {
         }
```

Comment:
executorService.shutdown does an orderly shutdown of the task. It does not wait for the tasks to be completed. So, the main thread after executing line 76 will go to line 79, and assertions will happen where the execution of tasks may or may not have got completed.

Requesting you to kindly change to https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadPoolExecutor.html#awaitTermination method which will wait for the tasks to be completed. Thanks.

Ref: https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadPoolExecutor.html#shutdown()

Reply:
aah. yes, will do that. we still need to cut the assertions about completed list and buffers returned, as any in-progress read will finish and add the buffer which will then stay until the threshold timeout